components/script/dom/webglprogram.rs

```diff
@@ -250,6 +293,25 @@ impl WebGLProgram {
         Ok(receiver.recv().unwrap())
     }
 
+    /// glGetProgramInfoLog
+    pub fn get_info_log(&self) -> WebGLResult<String> {
+        if self.is_deleted() {
+            return Err(WebGLError::InvalidOperation);
+        }
+        if self.link_called.get() {
+            let shaders_compiled = match (self.fragment_shader.get(), self.vertex_shader.get()) {
+                (Some(fs), Some(vs)) => fs.successfully_compiled() && vs.successfully_compiled(),
+                _ => false
+            };
+            if !shaders_compiled {
+                return Ok("One or more shaders failed to compile".to_string());
```

Comment:
Before linking a program getProgramInfoLog must return an empty string. After linking a program, if  GL_LINK_STATUS is false, getProgramInfoLog must return a string containing the error description.

We can't rely on the GPU getProgramInfoLog in all the scenarios because in some of them the real glLinkProgram function is never called, and getProgramInfoLog would return always empty string. For example if the WebGLProgram.link() function detects that a shader did not compile succesfully because Angle validation failed.

This error returns the expected message in the scenarios were the real glLinkProgram function has not been called